src/utils/registry.ts

```diff
@@ -385,22 +412,34 @@ export class Registry {
       // Remove stale browsers.
       await this._validateInstallationCache(linksDir);
 
-      // Install missing browsers for this package.
-      for (const browserName of browserNames) {
-        const revision = this._revision(browserName);
-        const browserDirectory = this.browserDirectory(browserName);
-        const title = `${browserName} v${revision}`;
-        const downloadFileName = `playwright-download-${browserName}-${hostPlatform}-${revision}.zip`;
-        await downloadBrowserWithProgressBar(title, browserDirectory, this.executablePath(browserName)!, this._downloadURL(browserName), downloadFileName).catch(e => {
-          throw new Error(`Failed to download ${title}, caused by\n${e.stack}`);
-        });
-        await fs.promises.writeFile(markerFilePath(browserDirectory), '');
+      // Install browsers for this package.
+      for (const executable of executables) {
+        if (executable._download)
+          await executable._download();
+        else
+          throw new Error(`ERROR: Playwright does not support installing ${executable.name}`);
```

Comment:
IIUC this will change later when channels move here too. Let me know if I misunderstand.

Reply:
Exactly.